monix-tail/shared/src/main/scala/monix/tail/Iterant.scala

```diff
@@ -1656,6 +1656,51 @@ sealed abstract class Iterant[F[_], A] extends Product with Serializable {
   final def scanEval[S](seed: F[S])(op: (S, A) => F[S])(implicit F: Sync[F]): Iterant[F, S] =
     IterantScanEval(self, seed, op)
 
+  /** Applies implicitly available `Monoid[B]` to combine elements using
+    * `Monoid[B]` identity value as the initial state and map every incoming
+    * element of this `Iterant` with provided function f: A => B.
```

Comment:
There's no "identity value" in Monoid, that value is called "empty" or "zero" (depending on the implementation).

Also the provided example defines a `Monoid` instance on the fly. That's not how this function will get used because people don't define `Monoid` instances on the fly like that — usually `Monoid` instances are attached to data types, with tests integrated with `cats-laws`, etc. What makes a `Monoid` a `Monoid` isn't only the interface, but the laws as well, which makes monoid instances expensive. So users that don't have a `Monoid[B]` instance already will probably not use this function.

I would say something like this:
```
Given a mapping function that returns a `B` type for which we have 
a [[cats.Monoid]] instance, returns a new stream that folds the incoming
elements of the sources using the provided `Monoid.combine`, with the
initial seed being the `Monoid.empty` value, emitting the generated values
at each step.

Equivalent with [[scan]] applied with the given [[cats.Monoid]], so given 
our `f` mapping function returns a `B`, this law holds:
{{{
  val B = implicitly[Monoid[B]]

  stream.scanMap(f) <-> stream.scan(B.empty)(B.combine)
}}}

Example:
{{{
  // Yields 2, 6, 12, 20, 30, 42
  Iterant[Task].of(1, 2, 3, 4, 5, 6).scanMap(x => x * 2)
}}}
```

Yeah, docs are harder than code 😛

Reply:
Regarding "identity value" - I see it often (maybe more in Haskell) but good point anyway because it might only confuse people. 

I've changed docs according to your suggestion. :)